src/algo/depth-to-rgb-calibration/calibration.cpp

```diff
@@ -0,0 +1,387 @@
+
+    copy_coefs( res );
+
+    return res;
+}
+
+calib calib::operator/( double factor ) const
+{
+    return (*this)*(1.f / factor);
+}
+
+calib calib::operator+( const calib & c ) const
+{
+    calib res;
+
+    for (auto i = 0; i < 9; i++)
+    {
+        res.k_mat.k_mat.rot[i] = k_mat.k_mat.rot[i] + c.k_mat.k_mat.rot[i];
+    }
+
+    for (auto i = 0; i < 9; i++)
+        res.rot.rot[i] = this->rot.rot[i] + c.rot.rot[i];
+
+    res.trans.t1 = this->trans.t1 + c.trans.t1;
+    res.trans.t2 = this->trans.t2 + c.trans.t2;
+    res.trans.t3 = this->trans.t3 + c.trans.t3;
+
+    copy_coefs( res );
+
+    return res;
+}
+
+calib calib::operator-( const calib & c ) const
+{
+    calib res;
+    for (auto i = 0; i < 9; i++)
+    {
+        res.k_mat.k_mat.rot[i] = k_mat.k_mat.rot[i] - c.k_mat.k_mat.rot[i];
+    }
+
+    for (auto i = 0; i < 9; i++)
+        res.rot.rot[i] = this->rot.rot[i] - c.rot.rot[i];
+
+    res.trans.t1 = this->trans.t1 - c.trans.t1;
+    res.trans.t2 = this->trans.t2 - c.trans.t2;
+    res.trans.t3 = this->trans.t3 - c.trans.t3;
+
+    copy_coefs( res );
+
+    return res;
+}
+
+calib calib::operator/( const calib & c ) const
+{
+    calib res;
+
+    for (auto i = 0; i < 9; i++)
+    {
+        res.k_mat.k_mat.rot[i] = k_mat.k_mat.rot[i] / c.k_mat.k_mat.rot[i];
+    }
+
+    for (auto i = 0; i < 9; i++)
+        res.rot.rot[i] = this->rot.rot[i] / c.rot.rot[i];
+
+
+    res.trans.t1 = this->trans.t1 / c.trans.t1;
+    res.trans.t2 = this->trans.t2 / c.trans.t2;
+    res.trans.t3 = this->trans.t3 / c.trans.t3;
+
+    copy_coefs( res );
+
+    return res;
+}
+
+
+p_matrix p_matrix::operator*(double step_size) const
+{
+    p_matrix res;
+
+    for (auto i = 0; i < 12; i++)
+        res.vals[i] = vals[i] * step_size;
+
+    return res;
+}
+
+p_matrix p_matrix::operator/(double factor) const
+{
+    return (*this)*(1.f / factor);
+}
+
+p_matrix p_matrix::operator+(const p_matrix & c) const
+{
+    p_matrix res;
+    for (auto i = 0; i < 12; i++)
+        res.vals[i] = vals[i] + c.vals[i];
+
+    return res;
+}
+
+p_matrix p_matrix::operator-(const p_matrix & c) const
+{
+    p_matrix res;
+    for (auto i = 0; i < 12; i++)
+        res.vals[i] = vals[i] - c.vals[i];
+
+    return res;
+}
+
+p_matrix p_matrix::operator*(const p_matrix & c) const
+{
+    p_matrix res;
+    for (auto i = 0; i < 12; i++)
+        res.vals[i] = vals[i] * c.vals[i];
+
+    return res;
+}
+
+p_matrix p_matrix::operator/(const p_matrix & c) const
+{
+    p_matrix res;
+    for (auto i = 0; i < 12; i++)
+        res.vals[i] = c.vals[i] == 0 ? 0 : vals[i] / c.vals[i];
+
+    return res;
+}
+
+double p_matrix::get_norma() const
+{
+    double grads_norm = 0;
+
+    for (auto i = 0; i < 12; i++)
+        grads_norm += vals[i] * vals[i];
+
+    grads_norm = sqrt(grads_norm);
+
+    return grads_norm;
+}
+
+static inline bool rtIsNaN( double value ) { return (value != value); }
+static inline bool rtIsInf( double value )
```

Comment:
Can be replaced by built-in std::isnan/isfinite/isnormal

Reply:
This is the Matlab Coder implementation (auto-generated)... not sure I want to touch it right now

Reply:
Thanks for clarification. No need for action